src/game/common/rts/team.cpp

```diff
@@ -0,0 +1,815 @@
+    }
+
+    if (m_productionConditionScript != nullptr) {
+        m_productionConditionScript->Delete_Instance();
+    }
+
+    m_productionConditionScript = nullptr;
+
+    for (int i = 0; i < ARRAY_SIZE(m_genericScripts); i++) {
+        if (m_genericScripts[i] != nullptr) {
+            m_genericScripts[i]->Delete_Instance();
+            m_genericScripts[i] = nullptr;
+        }
+    }
+}
+
+void TeamPrototype::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 2;
+    xfer->xferVersion(&version, 2);
+
+    int index;
+
+    if (xfer->Get_Mode() == XFER_SAVE) {
+        index = m_owningPlayer->Get_Player_Index();
+    }
+
+    xfer->xferInt(&index);
+
+    m_owningPlayer = g_thePlayerList->Get_Nth_Player(index);
+
+    if (version >= 2) {
+        xfer->xferAsciiString(&m_attackPriorityName);
+    }
+
+    xfer->xferBool(&m_productionConditionAlwaysFalse);
+    xfer->xferSnapshot(&m_teamTemplate);
+    unsigned short size = 0;
+    DLINK_ITERATOR<Team> iter = Iterate_Team_Instance_List();
+
+    while (!iter.Done()) {
+        size++;
+        iter.Advance();
+    }
+
+    xfer->xferUnsignedShort(&size);
+
+    if (xfer->Get_Mode() == XFER_SAVE) {
+        DLINK_ITERATOR<Team> iter2 = Iterate_Team_Instance_List();
+
+        while (!iter.Done()) {
+            Team *team = iter.Cur();
+            unsigned int id = team->Get_Team_ID();
+            xfer->xferUser(&id, sizeof(id));
+            xfer->xferSnapshot(team);
+            iter.Advance();
+        }
+    } else {
+        for (int i = 0; i < size; i++) {
+            unsigned int id;
+            xfer->xferUser(&id, sizeof(id));
+            Team *team = g_theTeamFactory->Find_Team_By_ID(id);
+
+            if (team == nullptr) {
+                team = g_theTeamFactory->Create_Team_On_Prototype(this);
+                team->Set_ID(id);
+            }
+
+            xfer->xferSnapshot(team);
+        }
+    }
+}
+
+void TeamPrototype::Team_About_To_Be_Deleted(Team *team)
+{
+    DLINK_ITERATOR<Team> iter = Iterate_Team_Instance_List();
+
+    while (!iter.Done()) {
+        unsigned int id;
+
+        if (team != nullptr) {
+            id = team->Get_Team_ID();
+        } else {
+            id = 0;
+        }
+
+        iter.Cur()->Remove_Override_Team_Relationship(id);
+        iter.Advance();
+    }
+}
+
+Team *TeamPrototype::Find_Team_By_ID(unsigned int id)
+{
+    DLINK_ITERATOR<Team> iter = Iterate_Team_Instance_List();
+
+    while (!iter.Done()) {
+        if (iter.Cur()->Get_Team_ID() == id) {
+            return iter.Cur();
+        }
+
+        iter.Advance();
+    }
+
+    return nullptr;
+}
+
+Team::Team(TeamPrototype *proto, unsigned int id) :
+    m_proto(proto),
+    m_teamID(id),
+    m_enteredOrExited(false),
+    m_active(false),
+    m_checkEnemySighted(false),
+    m_seeEnemy(false),
+    m_prevSeeEnemy(false),
+    m_wasIdle(false),
+    m_destroyThreshold(0),
+    m_curUnits(0),
+    m_unk1(false),
+    m_unk2(false)
+{
+    m_created = false;
```

Comment:
`Waypoint *m_currentWaypoint` not intialized here?

Reply:
fixed.